src/storage/src/source/postgres.rs

```diff
@@ -553,15 +553,22 @@ impl PostgresTaskInfo {
                 }));
 
                 self.row_sender.insert(mz_row.clone(), self.lsn).await;
+                // Failure scenario after we have produced at least one row, but before a
+                // successful `COMMIT`
+                fail::fail_point!("pg_snapshot_failure", |_| {
+                    Err(ReplicationError::Recoverable(anyhow::anyhow!(
+                        "recoverable errors should crash the process"
+                    )))
+                });
```

Comment:
I put this here instead of at the top of the function, to test if we actually recover after we have read rows, but haven't `COMMIT`'d yet, but I think it may cause a perf problems as its a read from a hashmap behind a rwlock...perhaps I can move it to a separate function I read once to populate a variable? or i can just move it back to the top of the function like before